### PR TITLE
[Feature] Support drop histogram statistics

### PR DESCRIPTION
**Pull request by HangyuanLiu**
## What type of PR is this：
- [ ] bug
- [x] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 8 / 72 (11.11%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|0|7|00.00%|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|3|00.00%|
|:large_blue_circle:|com/starrocks/statistic/StatisticSQLBuilder.java|0|3|00.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/CsvFileStatisticsStorage.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/statistic/StatisticExecutor.java|0|10|00.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/MockTpchStatisticStorage.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|0|5|00.00%|
|:large_blue_circle:|com/starrocks/sql/ast/DropHistogramStmt.java|0|9|00.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/EmptyStatisticStorage.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/CachedStatisticStorage.java|0|7|00.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/AnalyzeStmtAnalyzer.java|0|9|00.00%|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|1|7|14.29%|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|4|6|66.67%|
|:large_blue_circle:|com/starrocks/statistic/BasicStatsMeta.java|3|3|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 32 / 72 (44.44%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|3|00.00%|
|:large_blue_circle:|com/starrocks/statistic/StatisticSQLBuilder.java|0|3|00.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/CsvFileStatisticsStorage.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/statistic/StatisticExecutor.java|0|10|00.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/MockTpchStatisticStorage.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|0|5|00.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/EmptyStatisticStorage.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/CachedStatisticStorage.java|0|7|00.00%|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|1|7|14.29%|
|:large_blue_circle:|com/starrocks/sql/ast/DropHistogramStmt.java|7|9|77.78%|
|:large_blue_circle:|com/starrocks/sql/analyzer/AnalyzeStmtAnalyzer.java|8|9|88.89%|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|6|6|100.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|7|7|100.00%|
|:large_blue_circle:|com/starrocks/statistic/BasicStatsMeta.java|3|3|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 32 / 72 (44.44%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|3|00.00%|[271, 275, 279]|
|:large_blue_circle:|com/starrocks/statistic/StatisticSQLBuilder.java|0|3|00.00%|[107, 108, 109]|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/CsvFileStatisticsStorage.java|0|1|00.00%|[158]|
|:large_blue_circle:|com/starrocks/statistic/StatisticExecutor.java|0|10|00.00%|[113, 114, 117, 118, 119, 120, 121, 122, 123, 124]|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/MockTpchStatisticStorage.java|0|1|00.00%|[266]|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|0|5|00.00%|[418, 420, 421, 422, 424]|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/EmptyStatisticStorage.java|0|1|00.00%|[33]|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/CachedStatisticStorage.java|0|7|00.00%|[163, 164, 165, 166, 167, 168, 169]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|1|7|14.29%|[465, 820, 821, 823, 824, 825]|
|:large_blue_circle:|com/starrocks/sql/ast/DropHistogramStmt.java|7|9|77.78%|[35, 40]|
|:large_blue_circle:|com/starrocks/sql/analyzer/AnalyzeStmtAnalyzer.java|8|9|88.89%|[200]|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|7|7|100.00%|[]|
|:large_blue_circle:|com/starrocks/statistic/BasicStatsMeta.java|3|3|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 32 / 72 (44.44%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|3|00.00%|
|:large_blue_circle:|com/starrocks/statistic/StatisticSQLBuilder.java|0|3|00.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/CsvFileStatisticsStorage.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/statistic/StatisticExecutor.java|0|10|00.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/MockTpchStatisticStorage.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|0|5|00.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/EmptyStatisticStorage.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/CachedStatisticStorage.java|0|7|00.00%|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|1|7|14.29%|
|:large_blue_circle:|com/starrocks/sql/ast/DropHistogramStmt.java|7|9|77.78%|
|:large_blue_circle:|com/starrocks/sql/analyzer/AnalyzeStmtAnalyzer.java|8|9|88.89%|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|6|6|100.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|7|7|100.00%|
|:large_blue_circle:|com/starrocks/statistic/BasicStatsMeta.java|3|3|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 32 / 72 (44.44%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|3|00.00%|
|:large_blue_circle:|com/starrocks/statistic/StatisticSQLBuilder.java|0|3|00.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/CsvFileStatisticsStorage.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/statistic/StatisticExecutor.java|0|10|00.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/MockTpchStatisticStorage.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|0|5|00.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/EmptyStatisticStorage.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/CachedStatisticStorage.java|0|7|00.00%|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|1|7|14.29%|
|:large_blue_circle:|com/starrocks/sql/ast/DropHistogramStmt.java|7|9|77.78%|
|:large_blue_circle:|com/starrocks/sql/analyzer/AnalyzeStmtAnalyzer.java|8|9|88.89%|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|6|6|100.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|7|7|100.00%|
|:large_blue_circle:|com/starrocks/statistic/BasicStatsMeta.java|3|3|100.00%|

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/statistics/StatisticStorage.java [hunk 1]
```diff
@@ -15,6 +15,8 @@ public interface StatisticStorage {
 
     Map<ColumnRefOperator, Histogram> getHistogramStatistics(Table table, List<ColumnRefOperator> columns);
 
+    void expireHistogramStatistics(Long tableId, List<String> columns);
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/statistics/StatisticStorage.java:18]: Could make this method as a default method.

**Review by Youngwb — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 32 / 72 (44.44%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/ast/AstVisitor.java|0|3|00.00%|
|:large_blue_circle:|com/starrocks/statistic/StatisticSQLBuilder.java|0|3|00.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/CsvFileStatisticsStorage.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/statistic/StatisticExecutor.java|0|10|00.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/MockTpchStatisticStorage.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/sql/analyzer/PrivilegeChecker.java|0|5|00.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/CachedStatisticStorage.java|0|7|00.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/StatisticStorage.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|1|7|14.29%|
|:large_blue_circle:|com/starrocks/sql/ast/DropHistogramStmt.java|7|9|77.78%|
|:large_blue_circle:|com/starrocks/sql/analyzer/AnalyzeStmtAnalyzer.java|8|9|88.89%|
|:large_blue_circle:|com/starrocks/sql/analyzer/Analyzer.java|6|6|100.00%|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|7|7|100.00%|
|:large_blue_circle:|com/starrocks/statistic/BasicStatsMeta.java|3|3|100.00%|

**Review by kangkaisen — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Pull request merged**